### ISSUE TITLE
Unhandled enum in switch warnings

### ISSUE THREAD

**Issue by matzipan** [litehtml/litehtml#280]
```
In file included from out/include/litehtml/stylesheet.h:4,
                 from out/include/litehtml/element.h:7,
                 from out/include/litehtml/html_tag.h:4,
                 from out/include/litehtml/html.h:20,
                 from out/include/litehtml/litehtml.h:4,
                 from lib/litehtml_container/container.h:3,
                 from lib/litehtml_container/container.cpp:1:
out/include/litehtml/style.h: In destructor ‘litehtml::property_value::~property_value()’:
out/include/litehtml/style.h:88:32: warning: enumeration value ‘prop_type_invalid’ not handled in switch [-Wswitch]
   88 |                         switch (m_type)
      |                                ^
out/include/litehtml/style.h:88:32: warning: enumeration value ‘prop_type_inherit’ not handled in switch [-Wswitch]
out/include/litehtml/style.h:88:32: warning: enumeration value ‘prop_type_enum_item’ not handled in switch [-Wswitch]
out/include/litehtml/style.h:88:32: warning: enumeration value ‘prop_type_number’ not handled in switch [-Wswitch]
```

Are these valid issues?

**Comment by stasoid** [comment]
(I am answering because I wrote that code) No, these are not valid issues. Destructor must be called explicitly only if it is non-trivial. For `prop_type_invalid` and `prop_type_inherit` nothing is called because none of union values are used for these types. For `prop_type_enum_item` and `prop_type_number` nothing is called because `int` and `float` have trivial destructors. 
But I guess you can add those empty cases just to silence the warnings.

**Comment by matzipan** [comment]
Yes I was thinking more from the perspective of having empty cases just to satisfy the exhaustiveness check

**Comment by tordex** [comment]
Warnings should be fixed in this commit: d4453f5
